src/rgw/rgw_arn.cc

```diff
@@ -163,7 +163,7 @@ ARN::ARN(const std::string& resource_name, const std::string& type, const std::s
 }
 
 boost::optional<ARN> ARN::parse(const std::string& s, bool wildcards) {
-  static const std::regex rx_wild("arn:([^:]*):([^:]*):([^:]*):([^:]*):([^:]*)",
+  static const std::regex rx_wild("arn:([^:]*):([^:]*):([^:]*):([^:]*):(\\$\\{aws\\:[^:]*\\}|[^:]*)",
```

Comment:
the ARN formats from https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html are:
> arn:partition:service:region:account-id:resource-id
arn:partition:service:region:account-id:resource-type/resource-id
arn:partition:service:region:account-id:resource-type:resource-id

> Some resource identifiers include a parent resource (sub-resource-type/parent-resource/sub-resource) or a qualifier such as a version (resource-type:resource-name:qualifier).

if i'm understanding this regex, we're going to have trouble parsing any resources with colons in them. so i think this last match `([^:]*)` should be `(.*)` like in `rx_no_wild` below. then we don't need any extra support for these `${}` substitutions

Reply:
@adamemerson could you take a look at this one? do you remember why we needed this special pattern for `rx_wild`?

Reply:
I think you're right.